From 371534550579603eafa2501347c938908e0664f0 Mon Sep 17 00:00:00 2001
From: huangyifeng <huangyifeng@eswincomputing.com>
Date: Tue, 18 Mar 2025 15:17:45 +0800
Subject: [PATCH 327/540] fix:Fix cnoc npu timeout problem.

Changelogs:
On boards using the MPQ8785 as the NPU PMIC, performing continuous
PMIC power on/off operations may cause the output voltage to suddenly
drop to 0.7V. The exact reason for this issue is still unclear. This
patch provides a workaround by configuring the PMIC output twice
consecutively when enabling it, ensuring a stable output voltage

Signed-off-by: huangyifeng <huangyifeng@eswincomputing.com>
---
 drivers/regulator/mpq8785.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/regulator/mpq8785.c b/drivers/regulator/mpq8785.c
index a2bfd04af41b..01c9b3ac79a9 100644
--- a/drivers/regulator/mpq8785.c
+++ b/drivers/regulator/mpq8785.c
@@ -773,7 +773,7 @@ static int mpq8785_write(struct device *dev, enum hwmon_sensor_types type,
 	}
 	return ret;
 }
-static const struct hwmon_ops pac193x_hwmon_ops = {
+static const struct hwmon_ops mpq8785_hwmon_ops = {
 	.is_visible = mpq8785_is_visible,
 	.read = mpq8785_read,
 	.write = mpq8785_write,
@@ -781,7 +781,7 @@ static const struct hwmon_ops pac193x_hwmon_ops = {
 };
 
 static struct hwmon_chip_info mpq8785_chip_info = {
-	.ops = &pac193x_hwmon_ops,
+	.ops = &mpq8785_hwmon_ops,
 	.info = mpq8785_info,
 
 };
@@ -1067,10 +1067,6 @@ int mpq8785_regulator_enable(struct regulator_dev *rdev)
 {
 	struct i2c_client *client = to_i2c_client(rdev->dev.parent);
 	struct MPQ8785_DRIVER_DATA *data = i2c_get_clientdata(client);
-	u32 set_value = 0;
-	u32 get_value = 0;
-	int count = 0;
-	int ret;
 
 	dev_dbg(&rdev->dev, "%s.%d\n", __FUNCTION__, __LINE__);
 	return mpq8785_enable(&rdev->dev,data,  MPQ8785_MASK_OPERATION_ENABLE);
-- 
2.54.0

